modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/milker.dm

```diff
@@ -0,0 +1,319 @@
+		return FALSE
+	if(!iscarbon(being_milked)) // Trying to milk a robot? A megarachnid? *IAN*?
+		to_chat(milker, span_warning("You can't milk that!"))
+		return FALSE
+	if(being_milked.stat != CONSCIOUS)
+		to_chat(milker, span_warning("[self_suckle ? "You are in no condition to be milked!" : "[being_milked] is in no condition to be milked!"]"))
+		return FALSE
+	if(!istype(breasts))
+		to_chat(milker, span_warning("[self_suckle ? "You don't have any breasts!" : "[being_milked] doesn't seem to have any breasts!"]"))
+		return FALSE
+	if(!breasts.is_exposed())
+		to_chat(milker, span_warning("[self_suckle ? "You can't get to your breasts!" : "[being_milked]'s breasts aren't accessible!"]"))
+		return FALSE
+	if(!breasts.lactates)
+		to_chat(milker, span_warning("[self_suckle ? "You aren't lactating" : "[being_milked] doesn't seem to be lactating!"]"))
+		return FALSE
+	if(breasts.internal_fluids.total_volume <= 0)
+		to_chat(milker, span_warning("[self_suckle ? "You're out of milk!" : "[being_milked] is fresh out of milk!"]"))
+		return FALSE
+	if(!in_range(being_milked, milker))
+		to_chat(milker, span_warning("[self_suckle ? "Your breasts are too far away! ...somehow!" : "[being_milked]'s breasts are too far away!"]"))
+		return FALSE
+	return TRUE
+
+/// Checks if the drinker can, in fact, get their mouth onto the thing they're drinking
+/obj/item/milker/proc/mouth_check(mob/living/carbon/human/milker)
+	if(!ishuman(milker))
+		return FALSE
+	var/covered
+	if(milker.is_mouth_covered(head_only = 1))
+		covered = "headgear"
+	else if(milker.is_mouth_covered(mask_only = 1))
+		covered = "mask"
+	if(covered)
+		to_chat(milker, span_warning("You have to remove your [covered] first!"))
+		return FALSE
+	return TRUE
+
+/// Checks if the container is a container, unsealed, and has room. Retyrns a message if not!
+/obj/item/milker/proc/bottle_check(mob/living/carbon/being_milked, mob/living/carbon/human/milker, obj/item/reagent_containers/milk_bottle)
+	if(!istype(milk_bottle)) // Not a container?
+		return FALSE
+	if(!milk_bottle.is_open_container()) // Sealed container?
+		to_chat(milker, span_warning("[milk_bottle] is sealed!"))
+		return FALSE
+	if(milk_bottle.reagents?.holder_full()) // Is full?
+		to_chat(milker, span_warning("[milk_bottle] is full!"))
+		return FALSE
+	return TRUE
+
+/// Attempt to consume the contents of someone's breasts.
+/obj/item/milker/proc/drink_milk(mob/living/carbon/human/being_milked, mob/living/carbon/human/milker, obj/item/organ/genital/breasts/breasts)
+	if(!breast_check(being_milked, milker, breasts) || !mouth_check(milker))
+		return FALSE
+
+	var/self_suckle = (being_milked == milker) // Feeding off your own supply?
+
+	switch(handle_mode)
+		if(HANDLE_LIGHT) // Gentle paws...
+			if(self_suckle) // Drinking your own milk
+				if(milker.combat_mode)
+					milker.visible_message(
+					message = span_purple("[milker] yanks one of [milker.p_their()] nipples up to their mouth and starts nibbling..."),
```

Comment:
```suggestion
					message = span_purple("[milker] yanks one of [self_suckle ? "[milker.p_their()]" : "[being_milked]'s"] nipples up to their mouth and starts nibbling..."),
```
Just put the checks in-line to avoid repeated code here. You seem to have done this for the rest of the PR, so I'm not sure why it's done this way here?